client/src/pages/JoinGame.tsx

```diff
@@ -1,15 +1,42 @@
 import DisplayLogo from "../components/DisplayLogo";
-import { useState } from "react";
+import { useEffect, useState } from "react";
 
 import { socket } from "../App.tsx";
+import { useNavigate } from "react-router-dom";
+import { PLAYER_SCREEN } from "./pagePaths.ts";
 
 const JoinGame = () => {
+  const navigate = useNavigate();
+
   const [gameCode, setTournamentCode] = useState("");
   const [playerName, setPlayerName] = useState("");
 
+  const [loading, setLoading] = useState(false);
   const joinGame = () => {
+    console.log("trying to join");
     socket.emit("JOIN_GAME", gameCode, playerName);
   };
+  const [successfulJoin, setSuccessfulJoin] = useState(false);
```

Comment:
Can we change this to joinStatus or something. Which can have statuses of loading, success, invalidGameCode etc. Then we can remove the loading state, we won't need it.